logger/logger.go

```diff
@@ -0,0 +1,25 @@
+package logger
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+)
+
+type ctxKey int
+
+const loggerCtxKey ctxKey = 0
+
+// NewContext returns a new Context that carries a value l.
+func NewContext(ctx context.Context, l *zap.Logger) context.Context {
+	return context.WithValue(ctx, loggerCtxKey, l)
+}
+
+// FromContext returns the *zap.Logger value stored in ctx, if any.
+// If a value is not found, a nop logger is returned.
+func FromContext(ctx context.Context) *zap.Logger {
```

Comment:
Similar to 2 IMO but I think with loggers the interface is significantly less complex. It would also allow the called to store whatever logger it wants, event a mock one for tests or one that reports which we need to do with error logs

Reply:
Unfortunatley I think wrapping `zap` (or any other structured logger) would be **more** complex than a router (the number of functions on the [`Field`](https://pkg.go.dev/go.uber.org/zap#Field) type is daunting).

Reply:
Oh I'll check that out tomorrow, I'm unfamiliar with `Field`